drivers/avr/ws2812.c

```diff
@@ -171,12 +171,24 @@ void inline ws2812_setleds_rgbw(LED_TYPE *ledarray, uint16_t leds)
     I2C_Init();
     I2C_Start();
     I2C_Write(0x84);
-    uint16_t datlen = leds<<2;
-    uint8_t curbyte;
-    uint8_t * data = (uint8_t*)ledarray;
-    while (datlen--) {
-      curbyte=*data++;
-      I2C_Write(curbyte);
+    int i = 0;
+    #if defined(ERGODOX_LED_30)
```

Comment:
is there a different way to do this? what if you made this function weak and just overrode it in the ergodox keyboard? Feels wrong to have keyboard specific code in a driver.

Reply:
I agree that this is awful, but the actual driver itself seems to also be innately keyboard-specific, so far as I know -- it's specific to the ergodox EZ's weird i2c -> GPIO expander situation.